src/acl.rs

```diff
@@ -0,0 +1,269 @@
+            NoCommands => "nocommands".to_owned(),
+
+            AddPass(pass) => ">".to_owned() + pass,
+            RemovePass(pass) => "<".to_owned() + pass,
+            AddHashedPass(pass) => "#".to_owned() + pass,
+            RemoveHashedPass(pass) => "!".to_owned() + pass,
+            NoPass => "nopass".to_owned(),
+            ResetPass => "resetpass".to_owned(),
+
+            Pattern(pat) => "~".to_owned() + pat,
+            AllKeys => "allkeys".to_owned(),
+            ResetKeys => "resetkeys".to_owned(),
+
+            Reset => "reset".to_owned(),
+        };
+
+        out.write_arg(rule.as_bytes());
+    }
+}
+
+/// An info dictionary type storing Redis ACL information as multiple `Rule`.
+/// This type collects key/value data returned by the [`ACL GETUSER`][1] command.
+///
+/// [1]: https://redis.io/commands/acl-getuser
+#[derive(Debug, Eq, PartialEq)]
+pub struct AclInfo {
+    /// Describes flag rules for the user. Represented by [`Rule::On`][1],
+    /// [`Rule::Off`][2], [`Rule::AllKeys`][3], [`Rule::AllCommands`][4] and
+    /// [`Rule::NoPass`][5].
+    ///
+    /// [1]: ./enum.Rule.html#variant.On
+    /// [2]: ./enum.Rule.html#variant.Off
+    /// [3]: ./enum.Rule.html#variant.AllKeys
+    /// [4]: ./enum.Rule.html#variant.AllCommands
+    /// [5]: ./enum.Rule.html#variant.NoPass
+    pub flags: Vec<Rule>,
+    /// Describes the user's passwords. Represented by [`Rule::AddHashedPass`][1].
+    ///
+    /// [1]: ./enum.Rule.html#variant.AddHashedPass
+    pub passwords: Vec<Rule>,
+    /// Describes capabilities of which commands the user can call.
+    /// Represented by [`Rule::AddCommand`][1], [`Rule::AddCategory`][2],
+    /// [`Rule::RemoveCommand`][3] and [`Rule::RemoveCategory`][4].
+    ///
+    /// [1]: ./enum.Rule.html#variant.AddCommand
+    /// [2]: ./enum.Rule.html#variant.AddCategory
+    /// [3]: ./enum.Rule.html#variant.RemoveCommand
+    /// [4]: ./enum.Rule.html#variant.RemoveCategory
+    pub commands: Vec<Rule>,
+    /// Describes patterns of keys which the user can access. Represented by
+    /// [`Rule::Pattern`][1].
+    ///
+    /// [1]: ./enum.Rule.html#variant.Pattern
+    pub keys: Vec<Rule>,
+}
+
+impl FromRedisValue for AclInfo {
+    fn from_redis_value(v: &Value) -> RedisResult<Self> {
+        let values: Vec<Value> = FromRedisValue::from_redis_value(v)?;
+        let mut it = values.into_iter().skip(1).step_by(2);
+        let (flags, passwords, commands, keys) = match (it.next(), it.next(), it.next(), it.next())
+        {
+            (Some(flags), Some(passwords), Some(commands), Some(keys)) => {
+                // Parse flags
+                // Ref: https://git.io/JfNyb
+                let flag_values: Vec<Vec<u8>> = FromRedisValue::from_redis_value(&flags)?;
+                let mut flags = Vec::with_capacity(flag_values.len());
+                for flag in flag_values.into_iter() {
+                    let rule = match flag.as_slice() {
```

Comment:
```suggestion
                let flag_values = flags.as_sequence().ok_or_else(|| ..)?;
                let mut flags = Vec::with_capacity(flag_values.len());
                for flag in flag_values {
                    let flag = match flag {
                        Value::Data(flag) => flag,
                        _ => return Err("Expected bytes".into()),
                    };
                    let rule = match flag {
```

Reply:
Could even use `collect::<Result<_, _>>()` instead of using `with_capacity + push`